key_management/src/wallet.rs

```diff
@@ -0,0 +1,344 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::errors::Error;
+use super::{wallet_helpers, KeyInfo, KeyStore};
+use crate::MemKeyStore;
+use address::Address;
+use crypto::{Signature, SignatureType};
+use std::collections::HashMap;
+use std::str::FromStr;
+
+/// A Key, this contains a key_info, address, and public_key which holds the key type and private key
+#[derive(Clone, PartialEq, Debug, Eq)]
+pub struct Key {
+    key_info: KeyInfo,
+    // Vec<u8> is used because The public keys for BLS and SECP256K1 are not of the same type
+    public_key: Vec<u8>,
+    address: Address,
+}
+
+impl From<KeyInfo> for Key {
+    fn from(key_info: KeyInfo) -> Self {
+        let public_key =
+            wallet_helpers::to_public(*key_info.key_type(), key_info.private_key()).unwrap();
+        let address = wallet_helpers::new_address(*key_info.key_type(), &public_key).unwrap();
```

Comment:
if this is error prone, should probably be a `TryFrom` implementation, not a `From` so that the errors are handled and not just panic